api/logical.go

```diff
@@ -24,6 +24,20 @@ func (c *Logical) Read(path string) (*Secret, error) {
 	return ParseSecret(resp.Body)
 }
 
+func (c *Logical) List(path string) (*Secret, error) {
+	r := c.c.NewRequest("LIST", "/v1/"+path)
+	resp, err := c.c.RawRequest(r)
+	if resp != nil && resp.StatusCode == 404 {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
```

Comment:
I noticed this in the other methods as well, but I think even in a 404 we have to close the body. I looked at the Go impl. and I think there is a chance we're leaking stuff here.